src/west/app/project.py

```diff
@@ -1602,6 +1602,204 @@ def do_run(self, args, user_args):
+
+If the option is not set, "west grep" searches for the tool as follows:
+
+- git-grep: search for "git" (and run as "git grep")
+- ripgrep: search for "rg", then "ripgrep"
+- grep: search for "grep"
+
+TOOL ARGUMENTS
+--------------
+
+The "grep.<TOOL>-args" configuration options, if set, contain arguments
+that are always passed to the tool. The defaults for these are:
+
+- git-grep: (none)
+- ripgrep: (none)
+- grep: "--recursive"
+
+Additional command line arguments not recognized by "west grep" are
+passed to the tool after that. For example:
+
+  west grep --this-option-is-passed-to-the-tool
+
+To force an argument to be given to the tool instead of west, put
+it after a "--", like this:
+
+  west grep -- --project=val
+
+COLORS
+------
+
+By default, west will force the tool to print colored output as long
+as the "color.ui" configuration option is true. If color.ui is false,
+west forces the tool not to print colored output.
+
+Since all supported tools have similar --color options, you can
+override this behavior on the command line, for example with:
+
+  west grep --color=never
+
+To do this permanently, set "grep.color":
+
+  west config grep.color never
+'''
+
+class Grep(_ProjectCommand):
+
+    TOOLS = ['git-grep', 'ripgrep', 'grep']
+
+    DEFAULT_TOOL = 'git-grep'
+
+    DEFAULT_TOOL_ARGS = {
+        'git-grep': [],
+        'ripgrep': [],
+        'grep': ['--recursive'],
+    }
+
+    DEFAULT_TOOL_EXECUTABLES = {
+        # git-grep is intentionally omitted: use self._git
+        'ripgrep': ['rg', 'ripgrep'],
+        'grep': ['grep'],
+    }
+
+    def __init__(self):
+        super().__init__(
+            'grep',
+            'run grep or a grep-like tool in one or more local projects',
+            'Run grep or a grep-like tool in one or more local projects.',
+            accepts_unknown_args=True)
+
+    def do_add_parser(self, parser_adder):
+        parser = self._parser(parser_adder, epilog=GREP_EPILOG)
+        parser.add_argument('--tool', choices=self.TOOLS, help='grep tool')
+        parser.add_argument('--tool-path', help='path to grep tool executable')
+        # Unlike other project commands, we don't take the project as
+        # a positional argument. This inconsistency makes the usual
+        # use case of "search the entire workspace" faster to type.
+        parser.add_argument('-p', '--project', metavar='PROJECT',
+                            dest='projects', default=[], action='append',
+                            help='''project to run grep tool in (may be given
+                            more than once); default is all cloned active
+                            projects''')
+        return parser
+
+    def do_run(self, args, tool_cmdline_args):
+        tool = self.tool(args)
+        command_list = ([self.tool_path(tool, args)] +
+                        self.tool_args(tool, tool_cmdline_args))
+        failed = []
+        for project in self._cloned_projects(args,
```

Comment:
This is literally begging for parallelisation. `concurrent.futures` should make that trivial, example: https://github.com/thesofproject/sof/commit/9b2453e2366da640760c4d0a0b0d2094c9622978

Hopefully colorama+CMD.EXE will not get in the way again? (cf. #533 and friends)

Reply:
It's fast enough for me so can I assume that we can leave this to future work, or, better yet, for you to implement?

Better is the enemy of done, they say.

Reply:
Let's also keep in mind that:

- ripgrep is **already** parallelized, so running ripgrep in parallel is likely to over-commit your CPUs
- git grep has a `--threads` option and there is already a configuration option that people can use to use it by default

Reply:
> It's fast enough for me so can I assume that we can leave this to future work, or, better yet, for you to implement?

Yes, non-blocking comment of course. In fact I didn't use the "Request changes" option at all :-)